app/lib/app.dart

```diff
@@ -78,34 +79,47 @@ class _AppState extends State<App> with WidgetsBindingObserver {
           BlocProvider<NavigationCubit>(create: (context) => NavigationCubit()),
           BlocProvider<RegistrationCubit>(
               create: (context) => RegistrationCubit(coreClient: _coreClient)),
-          BlocProvider<ObservableUser>(
+          BlocProvider<LoadableUserCubit>(
             create: (context) =>
                 // loads the user on startup
-                ObservableUser((_coreClient..loadUser()).userStream),
+                LoadableUserCubit((_coreClient..loadUser()).userStream),
           ),
         ],
-        child: BlocListener<ObservableUser, UserState>(
-          listenWhen: (previous, current) =>
-              // only fire the side effect when the user logs in or out
-              (current.user == null || previous.user == null) &&
-              current.user != previous.user,
-          listener: (context, user) {
+        // This bloc has two tasks:
+        // 1. Listen to the loadable user and switch the navigation accordingly.
+        // 2. Provide the logged in user to the app, when it is loaded.
+        child: BlocConsumer<LoadableUserCubit, LoadableUser>(
+          listenWhen: _nullToSomeOrSomeToNull,
+          buildWhen: _nullToSomeOrSomeToNull,
+          listener: (context, loadableUser) {
             // Side Effect: navigate to the home screen or away to the intro
             // screen, depending on whether the user was loaded or unloaded.
-            switch (user) {
-              case LoadedUserState(user: final _?):
+            switch (loadableUser) {
+              case LoadedUser(user: final _?):
                 context.read<NavigationCubit>().openHome();
-              case LoadingUserState() || LoadedUserState(user: null):
+              case LoadingUser() || LoadedUser(user: null):
                 context.read<NavigationCubit>().openIntro();
             }
           },
-          child: router!,
+          builder: (context, loadableUser) => loadableUser.user != null
+              // Logged in user is accessible everywhere inside the app after
+              // the user is loaded
+              ? BlocProvider<UserCubit>(
+                  create: (context) =>
+                      UserCubit(coreClient: context.coreClient),
+                  child: router!,
+                )
+              : router!,
         ),
       ),
     );
   }
 }
 
+bool _nullToSomeOrSomeToNull(LoadableUser previous, LoadableUser current) =>
```

Comment:
I struggled to come up with a good name. Maybe you have an idea? Otherwise, I will add a comment. -- We want to check the the `LoadableUser` changed its `User` value either from non-null to null, of from null to non-null.

Reply:
Renamed to `_isUserLoadedOrUnloaded` and added a doc comment.